test/conversions.jl

```diff
@@ -45,3 +45,37 @@ end
         @test float32Val == Float32sr(float16Val)
     end
 end
+
+
+
+for SR in [Float16sr, BFloat16sr, Float32sr]
+
+    # corresponding FP type
+    FP = float(SR)
+
+    @eval begin
+        @testset "Stochastic conversion of Float64 to $(nameof($SR))" begin
```

Comment:
I don't think `@eval` is necessary.  This could simply be
```suggestion
@testset "Stochastic conversion of Float64 to $(nameof($SR))" for SR in (Float16sr, BFloat16sr, Float32sr)
    # corresponding FP type
    FP = float(SR)
    # ...
```

Reply:
Oh nice. I didn't know that we can add a for loop after the `testset` macro. Will change it in the next push